src/MongoDB.Driver.Core/Core/Compression/ICompressor.cs

```diff
@@ -0,0 +1,65 @@
+﻿/* Copyright 2019–present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+namespace MongoDB.Driver.Core.Compression
+{
+    /// <summary>
+    /// Represents the compressor type.
+    /// </summary>
+    public enum CompressorType
+    {
+        /// <summary>
+        /// The content of the message is uncompressed. This is realistically only used for testing.
+        /// </summary>
+        Noop = 0,
+        /// <summary>
+        /// The content of the message is compressed using snappy. NOT SUPPORTED YET.
+        /// </summary>
+        Snappy = 1,
+        /// <summary>
+        /// The content of the message is compressed using zlib. 
+        /// </summary>
+        Zlib = 2
+    }
+
+    /// <summary>
+    /// Represents a compressor.
+    /// </summary>
+    public interface ICompressor
+    {
+        /// <summary>
+        /// Gets the flag whether the compression is available to use or not.
+        /// </summary>
+        bool IsAvailable { get; }
```

Comment:
I believe this flag should be moved into the step with parsing `connectionString`. If the current solution is on the right way and if no objections, I will move this flag.

Reply:
Moved.